ql/src/java/org/apache/hadoop/hive/ql/exec/tez/YarnQueueHelper.java

```diff
@@ -101,8 +101,11 @@ public void checkQueueAccess(
         checkQueueAccessInternal(queueName, userName);
         return null;
       });
-    } catch (Exception exception) {
+    } catch (InterruptedException exception) {
+      // we need to catch only IO exception/Interrupted Exception here,
```

Comment:
the comment became outdated I believe:
```
we need to catch only IO exception/Interrupted
```
if we want to comment anything here, it should be inline with the current situation, which is, in terms of exceptions:
- InterruptedException: thrown by ugi.doAs, we don't want to handle it here
- IOException: thrown by ugi.doAs, we don't want to handle it here
- HiveException: thrown by checkQueueAccessInternal, we must throw it

also, please remove this:
```
// as part of [HIVE-27029](https://issues.apache.org/jira/browse/HIVE-27029) we are swallowing all exceptions and ignoring those causing yarn jobs are submitted
```
code comments should not contain what we were doing wrong in the past, no one cares about that :) a javadoc for the method is what we need, for example:
```
/**
* Checks yarn queue access of a given user for a given queue, and throws
* HiveException in case the user doesn't have access to the queue.
* @param  queueName the yarn queue name
* @param  userName the username
* @throws IOException when doAs throws an exception, we simply throw it
* @throws Interrupted when doAs throws an exception, we simply throw it
*/
```


regarding:
```
. now the user which does not have permission to access the queue, they cant run the yarn jobs. these changes we cant verify with UT.
```
this is not necessarily true, I think you can still mock a YarnQueueHelper, and check the following scenario:
- mock checkQueueAccessFromSingleRm to throw an IOException
- check if checkQueueAccess throws a HiveException

Reply:
@abstractdog fixed as per your suggestions.
This i tested manually and fix is working as per expectation.

Reply:
okay, I'm fine with the patch now, my point was whether we can write UT that shows this regression, like below, but you can do it in a follow-up ticket if you're interested
```
    YarnQueueHelper helper = spy(new YarnQueueHelper(new HiveConf()));
    doThrow(HiveException.class).when(helper).checkQueueAccessInternal(anyString(), anyString());
    try {
      helper.checkQueueAccess("fake", "fake");
      Assert.fail("HiveException is supposed to be thrown");
    } catch (Exception e) {
      Assert.assertEquals(HiveException.class, e.getCause().getClass());
    }
```